### PR TITLE
Added video compatibility

### PR DESCRIPTION
**Pull request by clementlamoureux**
# Added video compatibility

- first version of video compatibility
- format accepted : **mp4**, **webm**, **ogv** (because of native browser player)
- thumbnails based on first frames of video, maybe have to be reworked later
- full width/height on screen
- :warning: SQL Database changes

test
test

See also [https://github.com/LycheeOrg/Lychee-front/pull/5](https://github.com/LycheeOrg/Lychee-front/pull/5)

### CODE REVIEW

File: php/database/photos_table.sql [hunk 1]
```diff
@@ -8,7 +8,7 @@ CREATE TABLE IF NOT EXISTS `?` (
   `url` varchar(100) NOT NULL,
   `tags` varchar(1000) NOT NULL DEFAULT '',
   `public` tinyint(1) NOT NULL,
-  `type` varchar(10) NOT NULL,
+  `type` varchar(15) NOT NULL,
```
Comment (ildyria) [inline php/database/photos_table.sql]: This should not be placed here. Please create a migration file in `php/database` under the name `update_030200.php` to take care of that modification. Please have a look at https://github.com/LycheeOrg/Lychee/blob/master/php/database/update_030108.php for a template.

File: php/database/photos_table.sql [hunk 2]
```diff
@@ -20,7 +20,7 @@ CREATE TABLE IF NOT EXISTS `?` (
   `focal` varchar(20) NOT NULL,
   `takestamp` int(11) DEFAULT NULL,
   `star` tinyint(1) NOT NULL,
-  `thumbUrl` char(37) NOT NULL,
+  `thumbUrl` char(43) NOT NULL,
```
Comment (ildyria) [inline php/database/photos_table.sql]: this should not be placed here. See above.

File: php/Modules/Photo.php [hunk 3]
```diff
@@ -257,7 +257,7 @@ public function add(array $files, $albumID = 0, $returnOnError = false) {
 			$query  = Database::prepare(Database::get(), "INSERT INTO ? (id, title, url, description, tags, type, width, height, size, iso, aperture, make, model, shutter, focal, takestamp, thumbUrl, album, public, star, checksum, medium) VALUES ('?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?', '?')", $values);
 			$result = Database::execute(Database::get(), $query, __METHOD__, __LINE__);
 		} else {
-            $values = array(LYCHEE_TABLE_PHOTOS, $id, $info['title'], $photo_name, $file['type'], $info['size'], time(),  '../../play-icon.png', $albumID, $public, $star, $checksum, $medium);
+            $values = array(LYCHEE_TABLE_PHOTOS, $id, $info['title'], $photo_name, $file['type'], $info['size'], time(),  '../../'.$photo_name, $albumID, $public, $star, $checksum, $medium);
```
Comment (ildyria) [inline php/Modules/Photo.php]: Why is it a relative path `../../` ?

**Review by ildyria — CHANGES_REQUESTED** [review]
Really nice PR. :+1: 

Could you please update sql request (see bellow) so people already using Lychee can just do a `git pull` and everything works? Version number can be 030200. :)

**Comment by d7415** [comment]
This is great! The [FAQ](https://github.com/LycheeOrg/Lychee/wiki/FAQ#can-i-upload-videos) needs updating though :slightly_smiling_face:

**Comment by meminens** [comment]
I was able to upload an `mp4` file but the video is not playing.  See below.

image

macOS and Firefox (same in Safari)

Video codec info:

image

**Comment by ildyria** [comment]
@clementlamoureux Can you have a look ?

**Comment by d7415** [comment]
As a data point, this video (ffprobe output - looks similar) worked for me in Firefox and Chromium on Linux:
```
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'out.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf58.20.100
  Duration: 00:00:05.01, start: 0.000000, bitrate: 6952 kb/s
    Stream #0:0(eng): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1080x1920 [SAR 1:1 DAR 9:16], 6704 kb/s, 30 fps, 30 tbr, 15360 tbn, 60 tbc (default)
    Metadata:
      handler_name    : VideoHandle
    Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 256 kb/s (default)
    Metadata:
      handler_name    : SoundHandle
```

@bisherbas If you load the video directly (outside of Lychee) does the browser play it?

**Comment by clementlamoureux** [comment]
@bisherbas I think it's a browser compatiblity issue, as we are using native video element. I don't know what drivers are available on MacOS

**Comment by meminens** [comment]
`Refused to load media from 'https://xxx.com/lychee/uploads/big/1238912123kjhadasd981231.mp4' because it violates the following Content Security Policy directive: "default-src 'none'". Note that 'media-src' was not explicitly set, so 'default-src' is used as a fallback.`

It appears that this is the reason why video does not play. However, when I check my `apache2.conf` file I do not have a content security policy directive `"default-src 'none'"`. I am confused. HTTP header itself does not contain anything either.

Apache config:

`Header set Content-Security-Policy: "default-src 'self' media-src 'self' 'unsafe-inline' script-src 'self' 'unsafe-inline';"`

**Comment by d7415** [comment]
The .htaccess file sets it and has no media-src. I'll take a look later when I'm home unless someone beats me to it. Either add `media-src 'self';` there, delete/comment it, or change default-src from `'none'` to `'self'`

I should really set a matching CSP to test it properly, but I'm not using Apache, and am using my own CSP (mostly to set my report uri)

**Comment by meminens** [comment]
Thanks for pointing me to the `.htaccess` file. After changing `default-src` from `none` to `self`, the video started playing immediately.

**Pull request merged**